packages/compiler-core/src/parse.ts

```diff
@@ -101,7 +102,10 @@ function createParserContext(
   const options = extend({}, defaultParserOptions)
   for (const key in rawOptions) {
     // @ts-ignore
-    options[key] = rawOptions[key] || defaultParserOptions[key]
+    options[key] =
+      rawOptions[key] === undefined
```

Comment:
this should probably be
```suggestion
      !(key in rawOptions)
```

Reply:
Aha. Didn't notice that CI failed, fixed